### PR TITLE
Zach/404 page fix

### PR DESCRIPTION
**Pull request by zbayoff**
### What does this PR do?
1. Fixes 404 bug, navbar overflowing over sidenav. 
2. Updates 404 page layout and styles to match design of corpsite 404 as per: https://trello.com/c/POWL9jBi/2747-3-corpsite-and-docs-site-new-404-page-3

### Motivation
https://trello.com/c/zmAkLohn/3253-headers-crops-doc-page-in-404-doc-page

### Preview link
https://docs-staging.datadoghq.com/zach/404-page-fix/404.html

### CODE REVIEW

**Comment by l0k0ms** [comment]
Hello, Nice PR I have 2 feedback so far: 

* The search icon at the right side of the search bar seems to be broken: https://cl.ly/b3073d0b1ece
* The page search hasn't the same behavior as the navigation search (no modal displayed with the top 5 “suggested" links)

**Comment by zbayoff** [comment]
@l0k0ms 
1. The search icon should be fixed. 
2. Right, so on the corpsite, the search bar links to datadoghq.com/search/?s=aws
I wasn't sure about the functionality of the search for the docs site. What functionality would you prefer? Do you want the search bar to function the same as the 'search docs' bar in the sidenav?
If so, you also have to think about mobile, because the 404 search bar is visible on mobile.

**Comment by zbayoff** [comment]
@l0k0ms I talked to Eunmo from design and we think it might not make sense to have the 2nd search bar under the Error 404, it made more sense for the corpsite. So I have removed it.

File: layouts/404.html [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
         {{ partial "img.html" (dict "root" . "src" "sad_bits.png" "class" "img-fluid ml-auto" "alt" "sad") }}
+      </div>
+      <div class="col-xs-12 col-sm-8">
+        <h1 class="text-primary">Arf.</h1>
+        <h4 class="mt-0">It seems we can't find what you're looking for.</h4>
+        <h5 class="my-4 text-primary">Error 404</h5>
+        <p class="mt-2 font-semibold">Here are some helpful pages instead:</p>
+        <div class="helpful-pages-container m-b-4">
+          <ul class="ul-strip">
+            <li class="small"><a href="https://www.datadoghq.com/">Home</a></li>
```
Comment (l0k0ms) [inline layouts/404.html]:
```suggestion
            <li class="small"><a href="https://docs.datadoghq.com/">Home</a></li>
```

File: layouts/404.html [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
+            <li class="small"><a href="https://www.datadoghq.com/about/contact/">Contact Us</a></li>
+            <li class="small"><a href="https://docs.datadoghq.com/">Documentation</a></li>
```
Comment (l0k0ms) [inline layouts/404.html]: We can remove this link

File: layouts/404.html [hunk 3]
```diff
@@ -1,17 +1,29 @@
...
+      </div>
+      <div class="col-xs-12 col-sm-8">
+        <h1 class="text-primary">Arf.</h1>
+        <h4 class="mt-0">It seems we can't find what you're looking for.</h4>
+        <h5 class="my-4 text-primary">Error 404</h5>
+        <p class="mt-2 font-semibold">Here are some helpful pages instead:</p>
+        <div class="helpful-pages-container m-b-4">
+          <ul class="ul-strip">
+            <li class="small"><a href="https://www.datadoghq.com/">Home</a></li>
+            <li class="small"><a href="https://www.datadoghq.com/about/contact/">Contact Us</a></li>
```
Comment (l0k0ms) [inline layouts/404.html]:
```suggestion
            <li class="small"><a href="https://docs.datadoghq.com/help/">Contact Us</a></li>
```

**Pull request merged**